cmd/telegraf/telegraf.go

```diff
@@ -296,6 +316,63 @@ func (t *Telegraf) watchLocalConfig(signals chan os.Signal, fConfig string) {
 	signals <- syscall.SIGHUP
 }
 
+func (t *Telegraf) watchRemoteConfigs(signals chan os.Signal, interval time.Duration, remoteConfigs []string) {
+	configs := strings.Join(remoteConfigs, ", ")
+	log.Printf("I! Remote config watcher started for: %s\n", configs)
+
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	lastModified := make(map[string]string, len(remoteConfigs))
+	for {
+		select {
+		case <-signals:
+			return
+		case <-ticker.C:
+			for _, configURL := range remoteConfigs {
+				u, err := url.Parse(configURL)
+				if err != nil {
+					log.Printf("W! Error parsing config URL, %s: %s\n", configURL, err)
+					continue
+				}
+				rawQuery, err := config.AddHostParam(u)
+				if err != nil {
+					log.Printf("W! Error adding params to config URL, %s: %s\n", configURL, err)
+					continue
+				}
+				u.RawQuery = rawQuery
+				URLparam := u.String()
+
+				resp, err := http.Head(URLparam) //nolint: gosec // user provided URL
+				if err != nil {
+					log.Printf("W! Error fetching config URL, %s: %s\n", configURL, err)
+					continue
+				}
+				resp.Body.Close()
+
+				if resp.StatusCode != http.StatusOK {
+					log.Printf("E! Failed to fetch HTTP config: %s", resp.Status)
+					continue
+				}
+
+				modified := resp.Header.Get("Last-Modified")
+				if modified == "" {
+					log.Printf("E! Last-Modified header not found, stopping the watcher for %s\n", configURL)
+					delete(lastModified, configURL)
+				}
+
+				if lastModified[configURL] == "" {
+					lastModified[configURL] = modified
+				} else if lastModified[configURL] != modified {
```

Comment:
May be check that modified is really newer than last modified? In case of file behind CDN could be false triggered by older file

Reply:
Let's leave it to the future work if you don't mind, for now it's enough for us because we're not going to put this URL behind the CDN